array_test.go

```diff
@@ -115,6 +115,68 @@ func TestArray_Constructors(t *testing.T) {
 	})
 }
 
+func TestArray_Decode(t *testing.T) {
+	t.Run("Decode with slice of interface", func(t *testing.T) {
+		target := []interface{}{}
+		testValue := []interface{}{"Foo", 123.0}
+		reporter := newMockReporter(t)
+		arr := NewArray(reporter, testValue)
+		arr.Decode(&target)
+		arr.chain.assertNotFailed(reporter)
+		arr.Equal(target)
+	})
+	t.Run("Decode with slice of struct", func(t *testing.T) {
+		reporter := newMockReporter(t)
+		type S struct {
+			Foo int `json:"foo"`
+		}
+		testValue := []interface{}{
+			map[string]interface{}{
+				"foo": 123,
+			},
+			map[string]interface{}{
+				"foo": 456,
+			},
+		}
+		arr := NewArray(reporter, testValue)
+		var target []S
+		actualStruct := []S{{123}, {456}}
+		arr.Decode(&target)
+		arr.chain.assertFailed(reporter)
+		assert.Equal(reporter, actualStruct, target)
+	})
+	t.Run("Passing unmarshable value", func(t *testing.T) {
+		reporter := newMockReporter(t)
+		testValue := []interface{}{
+			map[string]interface{}{
+				"foo": 123,
+			},
+			map[string]interface{}{
+				"foo": 456,
+			},
+		}
+		arr := NewArray(reporter, testValue)
+		arr.Decode(123)
+		arr.chain.assertFailed(t)
+	})
+	t.Run("Target is nil", func(t *testing.T) {
+		reporter := newMockReporter(t)
+		testValue := []interface{}{"Foo", 123.0}
+		arr := NewArray(reporter, testValue)
+		arr.Decode(nil)
+		arr.chain.assertFailed(t)
+	})
+	t.Run("Empty interface", func(t *testing.T) {
+		reporter := newMockReporter(t)
+		testValue := []interface{}{"Foo", 123.0}
+		var target interface{}
+		arr := NewArray(reporter, testValue)
+		arr.Decode(&target)
+		arr.chain.assertNotFailed(reporter)
+		assert.Equal(reporter, testValue, target)
```

Comment:
It should be `assert.Equal(t`, not `assert.Equal(reporter`, because reporter is mock.